internal/populator-api/routes.go

```diff
@@ -100,16 +120,57 @@ func getSampleDataHandler(w http.ResponseWriter, r *http.Request, dbMongo *db.Mo
 		w.Write([]byte(err.Error()))
 		return
 	}
+
 	results, err := dbMongo.QueryMostRecent(r.Context(), limit)
 	if err != nil {
 		logger.Error(err.Error())
 		w.WriteHeader(http.StatusInternalServerError)
 		w.Write([]byte(err.Error()))
 		return
 	}
+
+	// Applying a host filter on the data if supplied
+
+	host := r.URL.Query().Get("host")
 	var viewModels []responses.DataViewModel
-	for _, i := range results {
-		viewModels = append(viewModels, models.ViewModelFromMongoRecord(i))
+	for _, record := range results {
+		// skip host filter if not supplied
+		if host == "" {
+			viewModels = append(viewModels, models.ViewModelFromMongoRecord(record))
+			continue
+		}
+		// Current approach is getting the dataRef by hashing
+		// the mongo record, then fetching annotations by that
+		// dataRef and finding their host
+		sampleData := models.SampleFromMongoRecord(record)
+		b, _ := json.Marshal(sampleData)
+		key := hashprovider.DeriveHash(b)
+
+		annotations, err := dbArango.QueryAnnotations(r.Context(), key)
```

Comment:
We'll need to trim the key here to avoid any issues with the db query.

Reply:
@michaelehab does this only happen in the get annotations query? I am curious as to why the `DeriveHash` method would give out a leading/trailing space.